templates/jenkins-slave-defaults.erb

```diff
@@ -54,9 +54,9 @@ TOOL_LOCATIONS="<%= @tool_locations -%>"
 
 DESCRIPTION="<%= @description -%>"
 
-# credentials
-JENKINS_USERNAME="<%= @ui_user -%>"
-JENKINS_PASSWORD="<%= @ui_pass -%>"
+# credentials should be single quoted
+JENKINS_USERNAME="'<%= @ui_user.gsub!(/^\'|\'?$/, '') -%>'"
```

Comment:
Yes you right exception on undef is really bad.
 
This regexp supposed to remove quotes if user input is already using quoted value as a workaround. So we could prevent double quoting.